OasisDEX/oasis-borrow
# 🧿 Switch between borrow and multiply UI across the same Ajna position

# Switch between borrow and multiply UI across the same Ajna position

Covers:
https://app.shortcut.com/oazo-apps/story/8838/switch-multiply-position-to-borrow-position
https://app.shortcut.com/oazo-apps/story/8839/switch-borrow-position-to-multiply-position

Both stories were lacking in design, I've used layouts and content from Maker as they were supposed to be similar, but I've adjusted look & feel to be closer to what we already had in Ajna. Second step of transition behaves similar to second step on transaction - dropdown is disabled and if users selects to go back in text button, they are taken back to first step.

[🌲-WBTC-USDC---Ajna---Borrow---Oasis-app-×-Ajna.webm](https://user-images.githubusercontent.com/16230404/234365402-3182072a-5f1e-4aa7-b92f-25e78a2933e6.webm)
  
## Changes 👷‍♀️

- Created new type of step in step manager for Ajna contest - "transition",
- added switch UI for Ajna Borrow position,
- created `useTransition` hook that handles entire process of transforming position from Borrow to Multiply and other way around - potentially that could be reused in other protocols,
- refined `getAjnaSidebarButtonsStatus`, `getAjnaSidebarPrimaryButtonActions` and `getPrimaryButtonLabelKey` to accommodate transition state,
  
## How to test 🧪

- Go to any Ajna Borrow positions you own and try changing it into Multiply position,
- page should reload with Multiply UI - at this point all data is fake and only working action is switching back to Borrow,
- try changing it into Borrow position,
- page should reload back to Borrow UI.
